### PR TITLE
[WeatherStatus] Support more weather codes, change icons, fix style

### LINKED ISSUE #30551: [weather_status] Snowfall
@eneiluj There is rainfall in "weather status". What about snowfall?

For meteorologists, these are two different types of precipitation.

### PR DESCRIPTION
**Pull request by julien-nc**
* Resolves #30551
* Closes #47702 

This branch is based on the one of #47702.

* Use icons provided by met.no at https://github.com/metno/weathericons
* Support more weather codes (a few snow ones)
* Fix location icon with dark theme
* Only use icons from `@mdi/svg`
* Fix alignment and color issues
* Make a difference between when no location has been set and when the weather code is unknown/unsupported

| Before | After |
|--------|--------|
| image | image |
| image | image |

### CODE REVIEW

File: apps/weather_status/src/App.vue [hunk 1]
```diff
@@ -495,103 +552,26 @@ export default {
...
-    mask: url(./../img/cross.svg) no-repeat;
-    mask-size: 18px 18px;
-    mask-position: center;
-    -webkit-mask: url(./../img/cross.svg) no-repeat;
-    -webkit-mask-size: 18px 18px;
-    -webkit-mask-position: center;
-    min-width: 44px !important;
-    min-height: 44px !important;
+.weather-image {
+	width: 28px;
```
Comment (susnux) [inline apps/weather_status/src/App.vue:570]: Magic value? Why 28px?
Our default icons are 16px (which is ~20px MDI icon size)

File: apps/weather_status/src/App.vue [hunk 2]
```diff
@@ -53,7 +73,10 @@
 					:aria-hidden="true"
 					@click="onFavoriteClick($event, favorite)">
 					<template #icon>
-						<IconStar :size="20" :class="{'favorite-color': address === favorite}" />
+						<NcIconSvgWrapper name="Star"
+							:svg="starSvg"
+							:size="20"
+							class="favorite-color" />
```
Comment (susnux) [inline apps/weather_status/src/App.vue:79]: Normally we do not use the favorite color in menus, it was used here to mark the current selected.
So this now changes the behavior as you can no longer see if the current address is a favorite.

Fine with me, but cc @nextcloud/designers

**Review by artonge — APPROVED** [review]

File: apps/weather_status/img/met.no.icons/clearsky_day.svg
Comment (susnux) [inline apps/weather_status/img/met.no.icons/clearsky_day.svg:1]: All the new files are missing the license information in `.reuse/dep5`.

Meaning you have to remove the part
```
Files: apps/weather_status/img/cloud-cloud.svg apps/weather_status/img/cloud-dots.svg apps/weather_status/img/cross.svg apps/weather_status/img/drop.svg apps/weather_status/img/fog.svg apps/weather_status/img/half-sun.svg apps/weather_status/img/heavy-rain.svg apps/weather_status/img/light-rain.svg apps/weather_status/img/moon-cloud-heavy-rain.svg apps/weather_status/img/moon-cloud-light-rain.svg apps/weather_status/img/moon-cloud-rain.svg apps/weather_status/img/moon-cloud.svg apps/weather_status/img/moon-small-cloud.svg apps/weather_status/img/moon.svg apps/weather_status/img/rain.svg apps/weather_status/img/snow.svg apps/weather_status/img/sun-cloud-heavy-rain.svg apps/weather_status/img/sun-cloud-light-rain.svg apps/weather_status/img/sun-cloud-rain.svg apps/weather_status/img/sun-cloud.svg apps/weather_status/img/sun-small-cloud.svg apps/weather_status/img/sun.svg apps/weather_status/img/thunder.svg apps/weather_status/img/umbrella.svg
Copyright: 2024 Designed by Freepik, titusurya <https://www.freepik.com/author/titusurya> at <https://www.freepik.com/free-vector/coloured-weather-icons-collection_895655.htm>
License: LicenseRef-freepikLicense
```

and replace with

```
Files: apps/weather_status/img/met.no.icons/*.svg
Copyright: 2015-2017 NRK og Meteorologisk institutt
License: MIT
```

**Review by susnux — CHANGES_REQUESTED** [review]
License information is important

File: apps/weather_status/src/App.vue [hunk 3]
```diff
@@ -6,36 +6,56 @@
...
+				<img v-else :src="weatherIconUrl" class="weather-image" alt="t('weather_status', 'Weather icon')">
+			</template>
 			<NcActionText v-if="gotWeather"
-				:aria-hidden="true"
-				:icon="futureWeatherIcon">
+				:aria-hidden="true">
+				<template #icon>
+					<NcLoadingIcon v-if="loading" />
+					<div v-else class="weather-action-image-container">
+						<img :src="futureWeatherIconUrl" class="weather-image" alt="t('weather_status', 'Future weather icon')">
```
[the remark this answers is not in the document]
Comment (julien-nc) [reply apps/weather_status/src/App.vue:21]: 🙈 👍

File: apps/weather_status/src/App.vue (hunk 2 above)
Comment (julien-nc) [reply apps/weather_status/src/App.vue:79]: Brought back the previous behaviour.

File: apps/weather_status/src/App.vue (hunk 1 above)
Comment (julien-nc) [reply apps/weather_status/src/App.vue:570]: I don't mind using 20px. It's just less visible.
I just raised the size as much as possible to make it easier to see/identify the icon. These weather icons are slightly more important than classic ones IMO.

Comment (susnux) [reply apps/weather_status/src/App.vue]: I agree, but then maybe:
```css
    width: calc(var(--default-clickable-area) - 2 * var(--default-grid-baseline));
```

File: apps/weather_status/src/App.vue [hunk 4]
```diff
@@ -6,36 +6,56 @@
...
+				<img v-else :src="weatherIconUrl" class="weather-image">
+			</template>
 			<NcActionText v-if="gotWeather"
-				:aria-hidden="true"
-				:icon="futureWeatherIcon">
+				:aria-hidden="true">
+				<template #icon>
+					<NcLoadingIcon v-if="loading" />
+					<div v-else class="weather-action-image-container">
+						<img :src="futureWeatherIconUrl" class="weather-image">
```
Comment (susnux) [inline apps/weather_status/src/App.vue:21]: Needs to be an empty alt tag to make it hidden in accessible tree:

```suggestion
						<img :src="futureWeatherIconUrl" alt="" class="weather-image">
```

File: apps/weather_status/src/App.vue [hunk 5]
```diff
@@ -6,36 +6,56 @@
 <template>
 	<div id="weather-status-menu-item">
 		<NcActions class="weather-status-menu-item__subheader"
-			:default-icon="weatherIcon"
 			:aria-hidden="true"
 			:aria-label="currentWeatherMessage"
 			:menu-name="currentWeatherMessage">
+			<template #icon>
+				<NcLoadingIcon v-if="loading" />
+				<img v-else :src="weatherIconUrl" class="weather-image">
```
Comment (susnux) [inline apps/weather_status/src/App.vue:14]:
```suggestion
				<img v-else :src="weatherIconUrl" alt="" class="weather-image">
```

File: apps/weather_status/src/App.vue [hunk 6]
```diff
@@ -6,36 +6,56 @@
...
+						:svg="addRemoveFavoriteSvg"
+						:size="20"
+						class="favorite-color" />
 				</template>
 				{{ addRemoveFavoriteText }}
 			</NcActionButton>
 			<NcActionSeparator v-if="address && !errorMessage" />
-			<NcActionButton icon="icon-crosshair"
+			<NcActionButton
 				:close-after-click="true"
```
Comment (susnux) [inline apps/weather_status/src/App.vue:51]:
```suggestion
			<NcActionButton :close-after-click="true"
```

(Sadly) our code style requires this

**Comment by susnux** [comment]
> License information is important

Thank you for fixing, `LicenseRef-freepikLicense` is now unused so you need to remove that from `/Licenses` directory :)

File: apps/weather_status/src/App.vue [hunk 7]
```diff
@@ -495,103 +551,26 @@ export default {
...
-    mask: url(./../img/cross.svg) no-repeat;
-    mask-size: 18px 18px;
-    mask-position: center;
-    -webkit-mask: url(./../img/cross.svg) no-repeat;
-    -webkit-mask-size: 18px 18px;
-    -webkit-mask-position: center;
-    min-width: 44px !important;
-    min-height: 44px !important;
+.weather-image {
+	width: 20px;
```
Comment (susnux) [inline apps/weather_status/src/App.vue:569]: If you prefer bigger icons:

```suggestion
	width: calc(var(--default-clickable-area) - 2 * var(--default-grid-baseline));
```

**Review by susnux — APPROVED** [review]

**Comment by jancborchardt** [comment]
@julien-nc very nice – just one detail: The previous icons worked a bit nicer with the style of Nextcloud, is there a specific reason they got replaced? If not I think it would be good to keep using the previous ones, if that’s ok with you? :)

**Pull request merged**